libraries/botbuilder-core/src/configurationBotFrameworkAuthentication.ts

```diff
@@ -0,0 +1,221 @@
+
+    /**
+     * Flag indicating whether or not to validate the address.
+     */
+    ValidateAuthority: t.String.Or(t.Boolean),
+
+    /**
+     * The Login URL used to specify the tenant from which the bot should obtain access tokens from.
+     */
+    ToChannelFromBotLoginUrl: t.String,
+
+    /**
+     * The Oauth scope to request.
+     *
+     * @remarks
+     * This value is used when fetching a token to indicate the ultimate recipient or `audience` of an activity sent using these credentials.
+     */
+    ToChannelFromBotOAuthScope: t.String,
+
+    /**
+     * The Token issuer for signed requests to the channel.
+     */
+    ToBotFromChannelTokenIssuer: t.String,
+
+    /**
+     * The OAuth URL used to get a token from OAuthApiClient.
+     */
+    OAuthUrl: t.String,
+
+    /**
+     * The OpenID metadata document used for authenticating tokens coming from the channel.
+     */
+    ToBotFromChannelOpenIdMetadataUrl: t.String,
+
+    /**
+     * The The OpenID metadata document used for authenticating tokens coming from the Emulator.
+     */
+    ToBotFromEmulatorOpenIdMetadataUrl: t.String,
+
+    /**
+     * A value for the CallerId.
+     */
+    CallerId: t.String,
+});
+
+/**
+ * Contains settings used to configure a [ConfigurationBotFrameworkAuthentication](xref:botbuilder-core.ConfigurationBotFrameworkAuthentication) instance.
+ */
+export type ConfigurationBotFrameworkAuthenticationOptions = t.Static<typeof TypedOptions>;
+
+const isRuntimeConfiguration = (val: unknown): val is Configuration => {
+    const config = val as Configuration;
+    return tests.isFunc(config.get) && tests.isFunc(config.set);
+};
+
+/**
+ * Creates a [BotFrameworkAuthentication](xref:botframework-connector.BotFrameworkAuthentication) instance from an object with the authentication values or a [Configuration](xref:botbuilder-dialogs-adaptive-runtime-core.Configuration) instance.
+ */
+export class ConfigurationBotFrameworkAuthentication extends BotFrameworkAuthentication {
+    private readonly inner: BotFrameworkAuthentication;
+
+    /**
+     * Initializes a new instance of the [ConfigurationBotFrameworkAuthentication](xref:botbuilder-core.ConfigurationBotFrameworkAuthentication) class.
+     *
+     * @param botFrameworkAuthConfig A [ConfigurationBotFrameworkAuthenticationOptions](xref:botbuilder-core.ConfigurationBotFrameworkAuthenticationOptions) object.
+     * @param credentialsFactory A [ServiceClientCredentialsFactory](xref:botframework-connector.ServiceClientCredentialsFactory) instance.
+     * @param authConfiguration A [Configuration](xref:botframework-connector.AuthenticationConfiguration) object.
+     * @param botFrameworkClientFetch A custom Fetch implementation to be used in the [BotFrameworkClient](xref:botframework-connector.BotFrameworkClient).
+     * @param connectorClientOptions A [ConnectorClientOptions](xref:botframework-connector.ConnectorClientOptions) object.
+     */
+    constructor(
```

Comment:
The more I see these complicated overloads with all the related type checking the less I like them. In .NET it makes sense since the runtime sorts it all out for you. In our world, it just complicates things. What do you think about adding static factory methods to these classes? The final constructor would expose just the lowest common denominator of options and the static factories could invoke the constructor accordingly. I think it'll clean things up significantly going forward and obviate the need for so much type checking in the overload-implementing function.